### PR TITLE
Create cs8515.md

### PR DESCRIPTION
**Pull request by balazsberes**
Added content for missing error message CS8515

## Summary

Describe your changes here.

Fixes #Issue_Number (if available)

### CODE REVIEW

File: docs/csharp/language-reference/compiler-messages/cs8515.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+description: "Compiler Error CS8515"
+title: "Compiler Error CS8515"
+ms.date: 24/10/2022
```
Comment (BillWagner) [inline docs/csharp/language-reference/compiler-messages/cs8515.md:4]: The tooling uses MM/DD/YYYY format:

```suggestion
ms.date: 10/24/2022
```

File: docs/csharp/language-reference/compiler-messages/cs8515.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+int x = 5;
+switch (x)
+{
+    case 5:
+        break;
+}
+```
+
+For more information, see [Switch](../language-reference/statements/selection-statements.md#the-switch-statement)
```
Comment (BillWagner) [inline docs/csharp/language-reference/compiler-messages/cs8515.md:43]: This link is incorrect:

```suggestion
For more information, see [Switch](../statements/selection-statements.md#the-switch-statement)
```

File: docs/csharp/language-reference/compiler-messages/cs8515.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+ The following sample generates CS8515:
+
+```csharp
+// CS8515.cs
+
+int x = 5;
+switch (x)
+    case 5:
+        break;
```
Comment (BillWagner) [inline docs/csharp/language-reference/compiler-messages/cs8515.md:25]: The example you've used generates a number of errors and warnings. I suggest the following to isolate only CS8515:

```suggestion
int x = 5;
switch x
{
    case 5:
        break;
}
```

File: docs/csharp/language-reference/compiler-messages/cs8515.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+```
+
+## To correct this error
+
+Add curly brackets: `{`  `}`:
```
Comment (BillWagner) [inline docs/csharp/language-reference/compiler-messages/cs8515.md:31]: Based on earlier comment:

```suggestion
Add parentheses: `(`  `)`:
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
Thanks for creating this new article @balazsberes 

I added comments on the new article.

There's also two other changes needed before we merge this:

First, this article needs to be added to the TOC.YML file in `docs/csharp/language-reference/compiler-messages`.

Second, `CS8515` needs to be removed from the list of warnings directed at `docs/csharp/misc/sorry-we-don-t-have-specifics-on-this-csharp-error.md`.

File: docs/csharp/language-reference/compiler-messages/cs8515.md (hunk 4 above)
Comment (balazsberes) [reply docs/csharp/language-reference/compiler-messages/cs8515.md:31]: I see, well I will add this one too as an example. Without the { } you will get also the 8515 error message, but you get other errors too: cs0118, cs1525, cs0219,cs1514, cs1513 AND cs8515 - tested with .NET 6 - Console App.

**Comment by balazsberes** [comment]
New PR: https://github.com/dotnet/docs/pull/32010

**Pull request closed without merging**